File: frontend/src/services/BankInformationService.js
import axios from "axios";

export default class BankInformationService{

    getAll(){
        return axios.get("http://localhost:8080/api/bankInformation/getAll")
    }

    getById(id){
        return axios.get(`http://localhost:8080/api/bankInformation/getById?id=${id}`)
    }

    getByUserId(userId){
        return axios.get(`http://localhost:8080/api/bankInformation/getByFkUserId?userId=${userId}`)
    }

    deleteById(id){
        return axios.delete(`http://localhost:8080/api/bankInformation/deleteById?id=${id}`)
    }

    add(values){
        return axios.post("http://localhost:8080/api/bankInformation/add", values)
    }

    update(values){
        return axios.patch("http://localhost:8080/api/bankInformation/update", values)
    }
}

File: backend/src/main/java/ecrow/backend/business/abstracts/BankInformationService.java
package ecrow.backend.business.abstracts;

import ecrow.backend.core.utilities.results.DataResult;
import ecrow.backend.core.utilities.results.Result;
import ecrow.backend.entities.concretes.BankInformation;
import ecrow.backend.entities.dtos.BankInformationAddDto;
import ecrow.backend.entities.dtos.BankInformationUpdateDto;

import java.util.List;

public interface BankInformationService {
    DataResult<List<BankInformation>> getAll();
    DataResult<BankInformation> getById(Integer id);
    DataResult<List<BankInformation>> getByFkUserId(Integer userId);
    Result deleteById(Integer id);

    Result add(BankInformationAddDto bankInformationAddDto);
    Result update(BankInformationUpdateDto bankInformationUpdateDto);
}
